ui/src/app/cohort-review/page-layout/page-layout.ts

```diff
@@ -1,31 +1,50 @@
-import {Component, OnInit} from '@angular/core';
+import {Component, OnDestroy, OnInit} from '@angular/core';
 
 import {cohortReviewStore} from 'app/cohort-review/review-state.service';
-import {navigate, urlParamsStore} from 'app/utils/navigation';
-import {ReviewStatus} from 'generated';
+import {currentCohortStore, currentWorkspaceStore, navigate, urlParamsStore} from 'app/utils/navigation';
+import {CohortReviewService, CohortsService, PageFilterRequest, PageFilterType, ParticipantCohortStatusColumns, ReviewStatus, SortOrder} from 'generated';
 
 @Component({
   templateUrl: './page-layout.html',
   styleUrls: ['./page-layout.css']
 })
-export class PageLayout implements OnInit {
-
-  subscription: any;
-  create = false;
-  constructor() {}
+export class PageLayout implements OnInit, OnDestroy {
+  reviewPresent: boolean;
+  cohortLoaded = false;
+  constructor(
+    private reviewAPI: CohortReviewService,
+    private cohortsAPI: CohortsService
+  ) {}
 
   ngOnInit() {
-    const review = cohortReviewStore.getValue();
+    const {ns, wsid, cid} = urlParamsStore.getValue();
+    const cdrid = +(currentWorkspaceStore.getValue().cdrVersionId);
+    this.reviewAPI.getParticipantCohortStatuses(ns, wsid, cid, cdrid, {
+      page: 0,
+      pageSize: 25,
+      sortColumn: ParticipantCohortStatusColumns.PARTICIPANTID,
+      sortOrder: SortOrder.Asc,
+      pageFilterType: PageFilterType.ParticipantCohortStatuses,
+    } as PageFilterRequest).subscribe(review => {
+      cohortReviewStore.next(review);
+      if (review.reviewStatus === ReviewStatus.NONE) {
+        this.reviewPresent = false;
+      } else {
+        this.reviewPresent = true;
+        navigate(['workspaces', ns, wsid, 'cohorts', cid, 'review', 'participants']);
+      }
+    });
+    this.cohortsAPI.getCohort(ns, wsid, cid).subscribe(cohort => {
+      currentCohortStore.next(cohort);
```

Comment:
I'm a bit concerned with this callsite pushing global state. SHould we consider moving this to the same location as the `currentWorkspaceStore` updates?

There are other components that depend on the `currentCohortStore` value, so now it feels like there some implicit dependencies on this component being rendered at the same time as all of those (it's also not clear in which component I can or cannot use `currentCohortStore`). Alternatively if this component is indeed meant to wrap anything with a cohort context, perhaps it should pass the cohort value down to its children as a property instead.

Reply:
It's good to call this out, since it's a bit outside the ordinary approach. Essentially, this component is now acting as the resolver, and is the sole manager of this state. And since it's a wrapper around all cohort-related pages, any component within that part of the hierarchy can expect to be able to connect to this store and get see up-to-date info. In that sense, it's actually relatively straightforward. A bigger issue, in my opinion, is the `cohortReviewStore`, which has multiple writers that push data at different times, and no clear owner. That stems from an underlying UX design issue, though.

Back to this store. Yes, we could just pass the cohort directly down to children, with one exception: the breadcrumb. Since that's rendered higher up in the tree, the only way for child components to affect it is via some sort of push mechanism. So we'd still need something like this, even if the rest of the data flow was made more standard.

To summarize: I think this is a reasonable refactoring step, since it addresses the underlying needs and has relatively low complexity. We can certainly continue to evolve this, and could also rethink the breadcrumb rendering in the future if desired.

Reply:
I see, this component is pretty much playing the same role as `with-workspace`. I was a bit thrown by the naming here since "cohort-review" doesn't really encompass every context where you might need to load/update a cohort (e.g. cohort clone page?) - though maybe it covers everything that exists today.

Could you capture this in a comment here describing how we would go about extending this approach and any current limitations? I can imagine a future dev being a bit lost figuring out how to add a new component which needs a cohort (and introducing bugs).